File: 287-find-the-duplicate-number/find-the-duplicate-number.py
# because size of array doest not exceede the size of n[i] for each elem we can check the index of each elem
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            temp = abs(nums[i])
            if(nums[temp] < 0) : return temp
            nums[temp] = -nums[temp]
        return -1
        

File: 287-find-the-duplicate-number/find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        for (int i = 0; i < nums.size(); i++){
            int store = abs(nums[i]);
            if (nums[store] < 0) return store;
            nums[store] = -nums[store];
        }
        return -1;
    }
};